### DISCUSSION TITLE
Training Coverage maps

### DISCUSSION THREAD

**Discussion by kevcfGit** [NVlabs/sionna#706] [Q&A]
Hi!
I have been working on a personal experiment where I plan to be able to optimize trainable parameters (such as material characteristic and antenna power) in order to characterize realistic coverage propagation for a very specific area of a city (small urban canyon).
For this purpose, I got a dataset of RSRP measurements and I have developed a training step based on NMSE where basically I take the following steps:
- Compute CoverageMap (CM) (5e7 ray samples // 2 runs)
- Exclude CM cells without coverage.
- Find CM cells with a sample (RSRP measurement) within a certain radius (as the CM cells are spatially static then in each step it finds the same RSRP sample).
- Apply the NMSE (per CM cell and then accumulate them).
![{BF0F1091-5E9A-40EC-AD0D-512180004A9D}](https://github.com/user-attachments/assets/96f25540-726f-4847-b106-55112b13d318)
Where:
	 ̅xi is the predicted value (the simulated RSS).
	xi  is the RSRP sample
        n is the number of CM cells that participated in the step.
- Apply gradients.

Also, it is important to point out that:
- Significant out layers from dataset have been removed.
- Spatial accuracy and space distribution of the samples have been verified with a GIS tool.

Then I have tried 2 approaches:
1. All roofs, all walls and ground in the scene share the same material within each group so they are train as a whole.
2. Each scene element has its own material so they are trained individually.

Both approaches start with recommended material parameters set by ITU and other inputs from the host network. 

Now, the problem is that the training in both approaches seems to stop improving the NMSE at a certain step, additionally, all the following NMSE results oscillate around this certain value and i get results similar to this:
![{A46283F9-FE8A-468A-B678-3AC1F831965A}](https://github.com/user-attachments/assets/946f4b4d-1248-4d01-9ea5-348415b070d0)

I understand this behavior as a result of my model reaching a critical point where it starts to "fight" itself in order to satisfy each CM cell trying to get to the target value (RSRP sample) but I am not certain of it; I have tried to give more freedom to the training (adding more trainable inputs) but in general, even though there is improvement, the training never seems to reach target.
Also, I reduced the scope of my experiment to one CM cell (any size) and in this scenario the training performs perfectly but as I increase the number of CM cells, the performance decays.

**I was wondering if you can provide any insights about my process**

I was very inspired by the "Learning of Material Properties via Gradient Descent" and particularly by the closing discussion which is what let me to work with coverage maps, I hope you can help me.

Thanks!

**Comment by SebastianCa** [comment]
Hi @kevcfGit,
please have a look at the [Instant Radio Maps repository](https://github.com/NVlabs/instant-rm). The [Calibration notebook](https://github.com/NVlabs/instant-rm/blob/main/notebooks/Calibration.ipynb) should be helpful for your problem.

**Reply by kevcfGit** [reply]
Thanks!! Is it always necessary to use instant radio maps for this type of calibrations or SIONNA is also capable? (I have been applying a similar logic in SIONNA and anyway it seems to no be capable of converging back to the optimal values for the material properties when using a synthetic ground truth as targets for RSS).
